### ISSUE TITLE
namespace detection when using 'oc run' vs 'docker run'

### ISSUE THREAD

**Issue by jmontleon** [ansible/ansible-container#395]
##### ISSUE TYPE
 - Documentation Report

We're trying to use ansible-container as a base for AnsibleApps used with Ansible Service Broker (names may change). 

We've noticed that the default behavior when invoking with 'docker run' and providing auth information for openshift the deployment is always performed in the project_name specified in the main.yml for each openshift object.

However, when using 'oc run' we're finding deployments are being performed in the project the meta-container is running in. We're trying to understand this behavior better and whether there is something we can pass as an environment variable to manipulate which namespace the containers are deployed to.

Would someone from the ansible-container team be able to help us understand the behavior?

**Comment by calfonso** [comment]
@chouseknecht thoughts on this?

**Comment by chouseknecht** [comment]
I would say at the moment that's a flaw in Ansible Container. We made the assumption that the Ansible Container project would be deployed to OpenShift using the base project name as the namespace or project name.

The `project_name`, which gets passed into the generated Ansible role [is set here](https://github.com/ansible/ansible-container/blob/develop/container/engine.py#L465) from the base project path.

In other words, if you create an Ansible Container project called `foo`, later when you run the `shipit` command `foo` becomes the OpenShift namespace or project name. 

That's definitely something we can fix in the coming release. Or, if you need it right now, I don't think it would be terribly difficult to add a command line option to `shipit`.
